packages/aws-cdk-lib/aws-eks/lib/cluster-resource.ts

```diff
@@ -96,7 +98,7 @@ export class ClusterResource extends Construct {
         // doesn't contain XXX key in object" (see #8276) by incrementing this
         // number, you will effectively cause a "no-op update" to the cluster
         // which will return the new set of attribute.
-        AttributesRevision: 2,
+        AttributesRevision: 3,
```

Comment:
for my learning, do we bump Rv each time when there is new attribute added?

Reply:
I believe so. It was introduced in this [PR](https://github.com/aws/aws-cdk/pull/8317).


>This also fixes https://github.com/aws/aws-cdk/issues/8276 in the following way: if a custom resource returns an attribute with an "undefined" value, CFN will fail with a "vendor response doesn't contain key" error. To avoid this, we return empty strings in case an attribute is undefined. This is also true for when adding new attributes, in which case updating to the new version will fail on previously deployed clusters with the same error. To mitigate this (and fix https://github.com/aws/aws-cdk/issues/8276 along the way), we add a fake property called "AttributesRevision" with a number that needs to be manually incremented every time new attributes are introduced. This will cause old clusters to be updated and the new attributes returned.